controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -178,18 +177,17 @@ func TestReconcileUpdateObservedGeneration(t *testing.T) {
 	g.Expect(result).To(Equal(ctrl.Result{}))
 
 	g.Eventually(func() int64 {
-		errGettingObject = testEnv.Get(ctx, util.ObjectKey(kcp), kcp)
-		g.Expect(errGettingObject).NotTo(HaveOccurred())
+		if err := testEnv.Get(ctx, util.ObjectKey(kcp), kcp); err != nil {
+			return -1
+		}
 		return kcp.Status.ObservedGeneration
-	}, 10*time.Second).Should(Equal(generation))
+	}, 10*time.Second).Should(BeNumerically(">=", generation))
```

Comment:
This avoids be too strict about reconciliation, given that a different controller could update the Generation, or get updated during reconciliation.